inference-engine/src/preprocessing/cpu_x86_avx2/ie_preprocess_gapi_kernels_avx2.cpp

```diff
@@ -144,6 +139,269 @@ void calcRowArea_32F(float dst[], const float *src[], const Size& inSz,
+            }
+        }
+
+        // horizontal pass
+        GAPI_DbgAssert(outSz.width >= half_nlanes);
+        for (int x = 0; x < outSz.width; ) {
+            for (; x <= outSz.width - half_nlanes && x >= 0; x += half_nlanes) {
+                v_int16 a10 = vx_load(&clone[4 * x]);
+                v_int16 a32 = vx_load(&clone[4 * (x + 4)]);
+                v_int16 a54 = vx_load(&clone[4 * (x + 8)]);
+                v_int16 a76 = vx_load(&clone[4 * (x + 12)]);
+
+                v_uint8 val_0;
+                v_uint8 val_1;
+                v_uint8 val_2;
+                v_uint8 val_3;
+
+                for (int c = 0; c < chanNum; ++c) {
+                    int shift = (half_nlanes / 4);
+
+                    v_gather_channel(val_0, tmp, mapsx, chanNum, c, x, 0);
+                    v_gather_channel(val_1, tmp, mapsx, chanNum, c, x, shift);
+                    v_gather_channel(val_2, tmp, mapsx, chanNum, c, x, shift * 2);
+                    v_gather_channel(val_3, tmp, mapsx, chanNum, c, x, shift * 3);
+
+                    v_int16 val0_0 = v_reinterpret_as_s16(v_expand_low(val_0));
+                    v_int16 val0_1 = v_reinterpret_as_s16(v_expand_low(val_1));
+                    v_int16 val0_2 = v_reinterpret_as_s16(v_expand_low(val_2));
+                    v_int16 val0_3 = v_reinterpret_as_s16(v_expand_low(val_3));
+
+                    v_int16 val1_0 = v_reinterpret_as_s16(v_expand_high(val_0));
+                    v_int16 val1_1 = v_reinterpret_as_s16(v_expand_high(val_1));
+                    v_int16 val1_2 = v_reinterpret_as_s16(v_expand_high(val_2));
+                    v_int16 val1_3 = v_reinterpret_as_s16(v_expand_high(val_3));
+
+                    v_int16 t0 = v_mulhrs(v_sub_wrap(val0_0, val1_0), a10);
+                    v_int16 t1 = v_mulhrs(v_sub_wrap(val0_1, val1_1), a32);
+                    v_int16 t2 = v_mulhrs(v_sub_wrap(val0_2, val1_2), a54);
+                    v_int16 t3 = v_mulhrs(v_sub_wrap(val0_3, val1_3), a76);
+
+                    v_int16 r0 = v_add_wrap(val1_0, t0);
+                    v_int16 r1 = v_add_wrap(val1_1, t1);
+                    v_int16 r2 = v_add_wrap(val1_2, t2);
+                    v_int16 r3 = v_add_wrap(val1_3, t3);
+
+                    v_uint8 q0 = v_packus(r0, r1);
+                    v_uint8 q1 = v_packus(r2, r3);                    
+
+                    v_uint8 q2 = v_shuffle_s8(q0, shuf_mask2);
+                    v_uint8 q3 = v_shuffle_s8(q1, shuf_mask2);
+
+                    v_uint8 q4 = v_blend_shiftleft<0xCC /*0b11001100*/, 4>(q2, q3);
+                    v_uint8 q5 = v_blend_shiftright<0xCC /*0b11001100*/, 4>(q2, q3);
+
+                    v_uint8 q6 = v256_permute4x64<0xD8>(q4);
+                    v_uint8 q7 = v256_permute4x64<0xD8>(q5);                    
+
+                    v_uint8 q8 = v_shuffle_s8(q6, shuf_mask3);
+                    v_uint8 q9 = v_shuffle_s8(q7, shuf_mask3);
+#if 0
```

Comment:
Please make it `#if - #else - #endif` also as you did in a vertical pass above

Reply:
Removed disabled snippet.